### DISCUSSION TITLE
Java Consumer stops consuming

### DISCUSSION THREAD

**Discussion by ton3r** [apache/pulsar#22313] [Q&A]
Hi, 

from time to time I see that my java consumer stops consuming or even get so slow that the queues increase dramatically.

Somehow I see that the consumer stuck somewhere in the segments and the "Mark delete position" and "Read position" hungs up and stay as if them are idle.

Restarting sometimes helps but mostly it starts consuming as slow as it was before the restart.

We initialize the consumer like

```java
            pulsarClient.newConsumer()
                    .topic(dataTopic)
                    .subscriptionName(dataSubscription)
                    .consumerName(consumerName + "_" + randomString(8))
                    .subscriptionType(SubscriptionType.Key_Shared)
                    .subscriptionInitialPosition(SubscriptionInitialPosition.Latest)
                    .messageListener(this::messageListener)
                    .startPaused(false)
                    .negativeAckRedeliveryBackoff(MultiplierRedeliveryBackoff.builder()
                            .minDelayMs(1000)
                            .maxDelayMs(60L * 1000)
                            .multiplier(2)
                            .build())
                    .ackTimeout(30, TimeUnit.SECONDS)
                    .subscribeAsync();
```

I have no idea to get this running back. Only clearing the queue completly get the things running again. But then I lose the information from the messages...

Please, have anyone any ideas on this?

--Tom

**Comment by lhotari** [comment]
What is your Pulsar broker and Pulsar client version? Unless you are running [a supported version of Pulsar broker and Pulsar client](https://pulsar.apache.org/contribute/release-policy/#supported-versions), I'd strongly suggest upgrading since a lot of bug fixes have been made to address similar problems that you are describing.

**Reply by ton3r** [reply]
We run version 3.1.2 as docker since a seemless upgrade to 3.2 failed with the docker image.

**Reply by lhotari** [reply]
> We run version 3.1.2 as docker since a seemless upgrade to 3.2 failed with the docker image.

There's Pulsar docker image 3.2.1 available too. Have you tested that?

How about the client version? Is that also 3.1.2 ? Please test with most recent supported version 3.2.1 or 3.0.3 .
